source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -746,26 +746,37 @@ def apply_final_journey_filters_post_finalize(response_list, request):
         _filter_similar_line_and_crowfly_journeys(journey_pairs_pool, request)
 
 
-def get_journey_pt_section(journey, criteria):
+def get_journey_extremity_pt_section(journey, criteria):
     if criteria == "arrival_stop_attractivity":
         sections = reversed(journey.sections)
     else:
         sections = journey.sections
-    extemity_pt_section = next((s for s in sections if s.type == response_pb2.PUBLIC_TRANSPORT), None)
+    extremity_pt_section = next((s for s in sections if s.type == response_pb2.PUBLIC_TRANSPORT), None)
 
-    assert extemity_pt_section
-    return extemity_pt_section
+    assert extremity_pt_section
+    return extremity_pt_section
 
 
 def get_journey_pt_extremity(journey, criteria):
-    extemity_pt_section = get_journey_pt_section(journey, criteria)
+    extemity_pt_section = get_journey_extremity_pt_section(journey, criteria)
 
     assert extemity_pt_section
     if criteria == "arrival_stop_attractivity":
         return extemity_pt_section.destination
     return extemity_pt_section.origin
 
 
+def compute_journey_virtual_duration(journey, criteria, virtual_fallbacks):
+    extremity_pt_section = get_journey_extremity_pt_section(journey, criteria)
+    extremity = get_journey_pt_extremity(journey, criteria)
```

Comment:
good catch... I have tried, but the code rapidly became a bowl of spaghetti, since `get_journey_extermity_pt_section` is also used elsewhere... since you bring it up, I have thought up another idea...

Reply:
Ah, I thought about something like this but was worried it would be awful. It's actually quite better than what I was worried about. Good for me :+1: